students/EricAdams/decorating_mailroom/decorated_mailroom.py

```diff
@@ -0,0 +1,348 @@
+#!/usr/bin/env python
+"""
+This is an object oriented version.
+Added decorators for monitoring use
+"""
+
+import sys
+import math
+from textwrap import dedent
+# import functools
+import datetime
+
+import make_donors
+
+# Utility so we have data to test with, etc.
+
+
+def monitor(method):
+    '''
+    Create a decorator to monitor each time a method is used
+    Log the method name, arguments and date/time used in a file
+    named monitor_mailroom.txt
+    '''
+    # @functools.wraps(method)
+
+    def log(*args, **kwargs):
+        result = method(*args, **kwargs)
+        name = method.__name__
+        arg_str = ', '.join(repr(arg) for arg in args)
+        kwarg_str = ', '.join(repr(kwarg) for kwarg in kwargs)
+        with open('monitor_mailroom.txt', 'a') as fopen:
```

Comment:
this is something to think about -- this re-opens the file each time -- could you open it once and keep a reference around for the duration of the program running?